modules/imgproc/src/convhull.cpp

```diff
@@ -297,15 +297,15 @@ void convexHull( InputArray _points, OutputArray _hull, bool clockwise, bool ret
         }
     }
 
-    if( !returnPoints )
+    if( !returnPoints ) {
         Mat(nout, 1, CV_32S, hullbuf).copyTo(_hull);
-    else
-    {
-        _hull.create(nout, 1, CV_MAKETYPE(depth, 2));
-        Mat hull = _hull.getMat();
-        size_t step = !hull.isContinuous() ? hull.step[0] : sizeof(Point);
-        for( i = 0; i < nout; i++ )
-            *(Point*)(hull.ptr() + i*step) = data0[hullbuf[i]];
+    } else {
+        _hull.create(nout, 1, _hull.type());
```

Comment:
`_hull.type()` may be undefined, e.g. in Python case of it output is just cv::Mat. The  original approach works better here.

Reply:
I am not sure how this should work without violating C++. we are not allowed to guess the size and layout of any object (after all a Point object is more than just 2 ints).